Automattic/vip
# Update NPM publish actions

## Description

Apply updates making the NPM publishing actions compatible with https://github.com/Automattic/vip-actions/pull/27. Splits the current `npm-publish` action into two actions: `npm-prepare-release` and `npm-publish`.

## Steps to Test

**Note:** Testing is not currently possible before merging https://github.com/Automattic/vip-actions/pull/27  and the current PR. The logic has been tested on another repository. 

Testing `npm-prepare-release`:

1. You can test `npm-prepare-release` by going to Actions -> `Prepare a new npm release` -> `Run workflow`. Ensure you select the `trunk` branch in the window that opens. Press `Run workflow`.
2. Verify that a new pull request appears similar to #1457.
3. Review and, if applicable, merge the pull request manually.

Testing `npm-publish`:

4. Verify that step 3 is complete.
5. Ensure that the package was published to NPM.
6. Ensure that a new release was published on GitHub.
7. Ensure another pull request appears that attempts to update the NPM package to the next develop version.
8. Review and merge the pull request.
